ubg/src/mafia/MafiaNight.js

```diff
@@ -1,12 +1,116 @@
-import React from 'react';
+import React, {useEffect} from 'react';
+import {useParams} from 'react-router-dom';
+import {
+  useFirestore,
+  useUser,
+  useFirestoreCollectionData,
+} from 'reactfire';
+import {makeStyles} from '@material-ui/core/styles';
+import Grid from '@material-ui/core/Grid';
+import Box from '@material-ui/core/Box';
+import Player from './Player';
+import PersonalInfo from './PersonalInfo';
+
+const useStyles = makeStyles((theme) => ({
+  main: {
+    height: '100vh',
+  },
+  video: {
+    background: theme.palette.primary.main,
+    textAlign: '-webkit-center',
+  },
+  gameContainer: {
+    display: 'flex',
+    flexDirection: 'column',
+    position: 'relative',
+  },
+  card: {
+    width: '100%',
+  },
+}));
 
 /**
  * @return {ReactElement} Mafia night element
  */
 export default function MafiaNight() {
+  const classes = useStyles();
+  const user = useUser();
+  const [initialize, setInitialize] = React.useState(false);
+  const [players, setPlayers] = React.useState([]);
+  const [userInfo, setUserInfo] = React.useState('');
+  const [roleText, setRoleText] = React.useState('');
+  const {roomId} = useParams();
+  const room = useFirestore().collection('rooms').doc(roomId);
+  const usersCollection = room.collection('users');
+  const usersData = useFirestoreCollectionData(usersCollection);
+
+  /**
+   * @return {undefined}
+   */
+  function loadNightData() {
+    if (initialize === false) {
+      setInitialize(true);
+      const allPlayers = [];
+      usersData.forEach(function(u) {
+        if (u.alive === true) {
+          allPlayers.push(u);
+        }
+        if (u.uid === user.uid) {
+          setUserInfo(u);
+          if ( u.role=== 1) {
+            setRoleText('Pretend to be clicking or thinking :)');
+          } else if (u.role === 2) {
+            setRoleText('Mafia, pick someone to kill.');
+          } else if (u.role === 3) {
+            setRoleText('Detective, who do you want to check tonight?');
+          } else if (u.role === 4) {
+            setRoleText('Doctor, who do you want to save tonight?');
+          }
+        }
+      });
+      setPlayers(allPlayers);
+    }
+  }
+  /**
+   * Load the all the mafia related data
+   */
+  useEffect(loadNightData, [initialize]);
```

Comment:
If you what to run this just when the component loads, I think you could try with the dependency array empty.